### PR TITLE
Related refactors to ctx.FormX functions

### PR DESCRIPTION
**Pull request by 6543**
followup of #16562
blocked by  #16571

### CODE REVIEW

**Comment by lunny** [comment]
No. This will not work. You still needs FormX for those Post requests. #16562 only the first step to do rename, next PRs we can add Query back but use ctx.URL for those GET requests.

**Comment by 6543** [comment]
I think we can do it in one ... and most func use Query in GET requests this few who doesnt can be migrated easyly within this pull

see discort maintainer channel :) @lunny

File: routers/api/v1/notify/user.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func ReadNotifications(ctx *context.APIContext) {
 	//     "$ref": "#/responses/empty"
 
 	lastRead := int64(0)
-	qLastRead := ctx.QueryTrim("last_read_at")
+	qLastRead := ctx.Req.FormValue("last_read_at")
```
Comment (lunny) [inline routers/api/v1/notify/user.go]: The original will trim the space.

File: routers/web/user/auth.go [hunk 2]
```diff
@@ -1643,16 +1642,16 @@ func ResetPasswdPost(ctx *context.Context) {
 	// Handle two-factor
 	regenerateScratchToken := false
 	if twofa != nil {
-		if ctx.QueryBool("scratch_code") {
-			if !twofa.VerifyScratchToken(ctx.QueryString("token")) {
+		if ctx.Req.FormValue("scratch_code") == "true" {
```
Comment (lunny) [inline routers/web/user/auth.go]: Should we check if the request is "true" or "1"?

**Comment by 6543** [comment]
@lunny now it's ready :)

**Comment by lunny** [comment]
CI is failing related

File: routers/web/repo/issue_label_test.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ func TestRetrieveLabels(t *testing.T) {
 		ctx := test.MockContext(t, "user/repo/issues")
 		test.LoadUser(t, ctx, 2)
 		test.LoadRepo(t, ctx, testCase.RepoID)
-		ctx.Req.Form.Set("sort", testCase.Sort)
+		ctx.Req.URL.RawQuery = "sort=" + testCase.Sort
```
Comment (lunny) [inline routers/web/repo/issue_label_test.go]: CI failure maybe related with this line change.

File: routers/web/user/home_test.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ func TestArchivedIssues(t *testing.T) {
 
 	ctx := test.MockContext(t, "issues")
 	test.LoadUser(t, ctx, 30)
-	ctx.Req.Form.Set("state", "open")
+	ctx.Req.URL.RawQuery = "state=open"
```
Comment (lunny) [inline routers/web/user/home_test.go]: It maybe contains other query parameters.

**Comment by 6543** [comment]
I should have tested it after rebase ... cleaned up relicts & untouch tests + fix goGet speedup

**Review by lunny — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**